Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -1682,18 +1698,15 @@ namespace AzToolsFramework
         auto poolManager = static_cast<AZ::InstancePoolManager*>(AZ::Interface<AZ::InstancePoolManagerInterface>::Get());
         auto handlerName = GetNameForHandlerId(handler.handlerId);
         auto handlerPool = poolManager->GetPool<PropertyHandlerWidgetInterface>(handlerName);
+
         if (handlerPool)
         {
             handlerPool->RecycleInstance(handler.handlerInterface);
         }
         else
         {
-            QTimer::singleShot(
-                0,
-                [interfacePointer = handler.handlerInterface]()
-                {
-                    delete interfacePointer;
-                });
+            // if there is no handler pool, then delete the handler immediately; parent widgets won't delete it twice
+            delete handler.handlerInterface;
```

Comment:
Shouldn't the handle.handlerInterface also be set to nullptr after deletion here?

Since the handler can be reused after the call, it would be better to make sure it compares to nullptr if it has been deleted.

```suggestion
            delete handler.handlerInterface;
            handler.handlerInterface = nullptr
```

Reply:
Good callout. Will do!